src/indexed_crate.rs

```diff
@@ -765,6 +773,39 @@ fn build_export_name_index(index: &HashMap<Id, Item>) -> HashMap<&str, &Item> {
     .collect()
 }
 
+fn build_variant_name_index(index: &HashMap<Id, Item>) -> HashMap<(Id, &str), (&Item, usize)> {
+    #[cfg(feature = "rayon")]
+    let iter = index.par_iter().map(|(_, value)| value);
+    #[cfg(not(feature = "rayon"))]
+    let iter = index.values();
+
+    iter.filter_map(|item| match &item.inner {
+        rustdoc_types::ItemEnum::Enum(e) => Some((item.id, e)),
+        _ => None,
+    })
+    .flat_map(|(id, enum_item)| {
+        #[cfg(feature = "rayon")]
+        let base_iter = enum_item.variants.par_iter();
+        #[cfg(not(feature = "rayon"))]
+        let base_iter = enum_item.variants.iter();
+
+        base_iter.copied().enumerate().map(move |(i, var_id)| {
+            let var = index.get(&var_id).expect("Variant id should be correct.");
+            (
+                (
+                    id,
+                    var.name
+                        .as_ref()
+                        .expect("Variant should have a name.")
+                        .as_str(),
+                ),
+                (var, i),
+            )
```

Comment:
This is very difficult to read.

Please consider giving each intermediate result an intuitive name, and then just using the return expression to construct the tuples: `((enum_id, variant_name), (variant_id, index))` or similar.

In general, I'm not a fan of using super short variable names like `var`. They save a split second over using something more descriptive like `variant_item`, but the code is then much more difficult to re-read and re-understand 6 months or 2 years later. So whenever possible, please make the code not just correct on day 1 but easy to understand and extend 2 years later!